cmake/GetVersionFromGit.cmake

```diff
@@ -0,0 +1,120 @@
+set(VERSION_REGEX "?([0-9]+\\.[0-9]+\\.[0-9]+)" CACHE STRING "Regex for extracting version")
+set(ARCHIVAL_FILE "${CMAKE_SOURCE_DIR}/.git_archival.txt" CACHE STRING "Path to .git_archival.txt")
+set(DESCRIBE_NAME_KEY "describe-name: " CACHE STRING "Key for describe name in .git_archival.txt")
+set(COMMIT_HASH_KEY "commit: " CACHE STRING "Key for commit hash in .git_archival.txt")
+
+
+# Combine prefix and regex
+set(VERSION_REGEX_WITH_PREFIX "^${VERSION_PREFIX}${VERSION_REGEX}")
+
+# Find Git
+find_package(Git)
+
+# Attempt to retrieve version from Git
+if(EXISTS "${CMAKE_SOURCE_DIR}/.git" AND GIT_FOUND)
+    message(STATUS "Using git describe for versioning")
+
+    # Extract the version string
+    execute_process(
+        COMMAND git describe --tags --dirty
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        OUTPUT_VARIABLE VERSION_STRING
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_QUIET
+    )
+
+    # If no tags are found, set version to 0 and show a warning
+    if(VERSION_STRING STREQUAL "")
+        set(VERSION_STRING "0.0.0")
+        message(WARNING 
+            "No git tags found. Version set to 0.0.0.\n"
+            "Run 'git fetch --tags' to ensure proper versioning.\n"
+            "For more information, see OpenMC developer documentation."
+        )
+    endif()
+
+    # Extract the commit hash
+    execute_process(
+        COMMAND git rev-parse HEAD
+        WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+        OUTPUT_VARIABLE COMMIT_HASH
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+else()
+    message(STATUS "Using archival file for versioning: ${ARCHIVAL_FILE}")
+    if(EXISTS "${ARCHIVAL_FILE}")
+        file(READ "${ARCHIVAL_FILE}" ARCHIVAL_CONTENT)
+
+        # Extract the describe-name line
+        string(REGEX MATCH "${DESCRIBE_NAME_KEY}([^\\n]+)" VERSION_STRING "${ARCHIVAL_CONTENT}")
+        if(VERSION_STRING MATCHES "${DESCRIBE_NAME_KEY}(.*)")
+            set(VERSION_STRING "${CMAKE_MATCH_1}")
+        else()
+            message(FATAL_ERROR "Could not extract version from ${ARCHIVAL_FILE}")
+        endif()
+
+        # Extract the commit hash
+        string(REGEX MATCH "${COMMIT_HASH_KEY}([a-f0-9]+)" COMMIT_HASH "${ARCHIVAL_CONTENT}")
+        if(COMMIT_HASH MATCHES "${COMMIT_HASH_KEY}([a-f0-9]+)")
+            set(COMMIT_HASH "${CMAKE_MATCH_1}")
+        else()
+            message(FATAL_ERROR "Could not extract commit hash from ${ARCHIVAL_FILE}")
+        endif()
+    else()
+        message(FATAL_ERROR "Neither git describe nor ${ARCHIVAL_FILE} is available for versioning.")
+    endif()
+endif()
+
+# Ensure version string format
+if(VERSION_STRING MATCHES "${VERSION_REGEX_WITH_PREFIX}")
+    set(VERSION_NO_SUFFIX "${CMAKE_MATCH_1}")
+else()
+    message(FATAL_ERROR "Invalid version format: Missing base version in ${VERSION_STRING}")
+endif()
+
+# Check for development state
+if(VERSION_STRING MATCHES "-([0-9]+)-g([0-9a-f]+)")
```

Comment:
Isn't this where you would use `VERSION_SUFFIX`? I notice it isn't used anywhere...

Reply:
Hmmm... that was more of a question and something to think about because:
• the regex you replaced is different from the one in SUFFIX
• the you probably need to replace the regex on line 96 to match (one of the advantages of using a variable)

So which regex is correct?